src/cnab.rs

```diff
@@ -69,24 +67,64 @@ impl Bundle {
     /// let bundle = Bundle::from_file("testdata/bundle.json").unwrap();
     /// assert_eq!(bundle.name, "helloworld");
     /// ```
-    pub fn from_file<P: AsRef<Path>>(path: P) -> Result<Self, BundleParseError> {
-        let file = File::open(path)?;
-        Self::from_json(file)
-    }
-
-    /// Deserialize a `Bundle` from any type implementing `Read`.
-    pub fn from_json<R: Read>(reader: R) -> Result<Self, BundleParseError> {
-        let bundle = serde_json::from_reader(reader)?;
-        Ok(bundle)
+    pub fn from_file(file_path: &str) -> Result<Self, BundleParseError> {
+        let file = File::open(Path::new(&file_path))?;
```

Comment:
Keeping the function signature the same here allows for more flexibility (and also provides some additional direction to clients based on the type). This is the [same approach](https://github.com/rust-lang/rust/blob/61d286e9d0530bea8fb1ae24be3989baa9ea08eb/src/libstd/fs.rs#L369) taken by `std::fs::File::open`.

> In public interfaces, you usually don't want to use `Path` or `PathBuf` directly, but rather a generic `P: AsRef<Path>` or `P: Into<PathBuf>`. That way the caller can pass in `Path`, `PathBuf`, `&str` or `String`. ([source](https://www.reddit.com/r/rust/comments/7mu7q1/is_working_with_paths_always_this_painful/drwoat0))